src/common/dom/apply_themes_on_element.ts

```diff
@@ -20,27 +21,34 @@ const hexToRgb = (hex: string): string | null => {
  *
  * element: Element to apply theme on.
  * themes: HASS Theme information
- * localTheme: selected theme.
- * updateMeta: boolean if we should update the theme-color meta element.
+ * selectedTheme: selected theme.
  */
 export const applyThemesOnElement = (
   element,
-  themes,
-  localTheme,
-  updateMeta = false
+  themes: HomeAssistant["themes"],
+  selectedTheme?: string
 ) => {
+  const newTheme = selectedTheme ? themes.themes[selectedTheme] : undefined;
+
+  // Styles that need to be reset from the previous theme
   if (!element._themes) {
+    if (!newTheme) {
+      // No styles to reset, and no styles to set
+      return;
+    }
     element._themes = {};
   }
-  let themeName = themes.default_theme;
-  if (localTheme === "default" || (localTheme && themes.themes[localTheme])) {
-    themeName = localTheme;
-  }
+
+  // Add previous set keys to reset them
   const styles = { ...element._themes };
-  if (themeName !== "default") {
-    const theme = { ...derivedStyles, ...themes.themes[themeName] };
+  if (newTheme) {
+    const theme = {
+      ...derivedStyles,
+      ...newTheme,
+    };
     Object.keys(theme).forEach((key) => {
```

Comment:
Could we change to a `for (const key of Object.keys(theme))` and use `continue` instead of `return`.  It's cheaper to run as we don't have to execute extra functions.